stub-idp/src/main/java/uk/gov/ida/stub/idp/views/SingleIdpPromptPageView.java

```diff
@@ -36,4 +50,48 @@ public URI getVerifySubmissionUrl() {
     public UUID getUniqueId() {
         return uniqueId;
     }
+
+    public DatabaseIdpUser getIdpUser() { return idpUser; }
+
+    public String getFirstName() {
+        if (!idpUser.getFirstnames().isEmpty()) {
+            return idpUser.getFirstnames().get(0).getValue();
+        }
+        return createEmptySimpleMdsStringValue().getValue();
+    }
+
+    private MatchingDatasetValue<String> createEmptySimpleMdsStringValue() {
+        return new MatchingDatasetValue<>("", null, null, true);
+    }
+
+    public String getSurname() {
+        Collection<String> surnameValues = Collections2.transform(this.idpUser.getSurnames(), new Function<MatchingDatasetValue<String>, String>() {
```

Comment:
Would have thought this and `getFirstName()` would follow a similar pattern ... this seems overly complex given the dataset.

Reply:
I copied this out of the ConsentResource assuming there was a reason for it.  Originally I factored it out into separate class but for some reason thought that I no longer used it twice so removed the class.  I've now refactored it again so the methods only appear in one class and the using classes delegate to that.  It appears that there's a deliberate intention to only display the first first-name but all surnames.

Is there a requirement to display all surnames? (married- and maiden-names I presume to be the most obvious cases) or should it be simplified?